modules/configuration/main.tf

```diff
@@ -32,14 +32,13 @@ resource "azurerm_key_vault_secret" "graphdb-license" {
 }
 
 resource "azurerm_key_vault_secret" "graphdb-cluster-token" {
-  count = var.graphdb_java_options != null ? 1 : 0
-
   key_vault_id = data.azurerm_key_vault.graphdb.id
 
   name  = var.graphdb_cluster_token_name
   value = base64encode(local.graphdb_cluster_token)
 
-  tags = var.tags
+  tags       = var.tags
+  depends_on = [random_password.graphdb-cluster-token]
```

Comment:
Wouldn't every `terraform apply` generate a new cluster token if I don't set the variable?

Reply:
Should be tested, but I think if such entry exists in the key vault it won't be created anew.

Reply:
I've been using it without setting a token and it was not regenerated on apply, AFAIK

Reply:
@viktor-ribchev this `depends_on` would break when the token is provided, also, I don't think it is needed.